### DISCUSSION TITLE
Chem.MolToSmiles() output is not consistent with the documented description(canonical=True)

### DISCUSSION THREAD

**Discussion by ShangChien** [rdkit/rdkit#6057] [Q&A] [answered]
#### I want optimize my code, which enumerate molecule according smiles (like Greg Landrum's blog: [Intro to the molecule enumerator](https://greglandrum.github.io/rdkit-blog/posts/2021-05-13-intro-to-the-molecule-enumerator.html)).
The smiles from official `rdMolEnumerator` always has an extra symbol `-` , after pass bundle[0] to `Chem.MolToSmiles(mol, canonical=True)`, the result like this : `'c1cnc-c2cnccc2-1'`
the problem is that `'c1cnc-c2cnccc2-1'` is not a canonical smiles, it is necessary to call `Chem.CanonSmiles('c1cnc-c2cnccc2-1')`, then get the real canonical smiles: `'c1cc2ccncc2cn1'`
#### reproduce:
version=rdkit.2022.3.5
```py
m=Chem.MolFromSmiles('c1ccncc1.*:ccnc:* |m:6:0,11:5|',sanitize=False)
bundle=rdMolEnumerator.Enumerate(m)
smi=Chem.MolToSmiles(bundle[0],canonical=True)
smi ## 'c1cnc-c2cnccc2-1'
```
```py
canonical_smi=Chem.CanonSmiles(smi)
canonical_smi ## 'c1cc2ccncc2cn1'
```
expectation: `smi==canonical_smi` is True

**Comment by jrkarki** [comment]
Hi there,

One solution you could try is to convert your smiles to structure and back to smiles. As explained in rdkit doc, it is as follows:

Chem.MolToSmiles(Chem.MolFromSmiles('C1=CC=CN=C1'))
'c1ccncc1'
Chem.MolToSmiles(Chem.MolFromSmiles('c1cccnc1'))
'c1ccncc1'
Chem.MolToSmiles(Chem.MolFromSmiles('n1ccccc1'))
'c1ccncc1'

Let me know how effective you find it. 

Best,
Reagon

**Comment by greglandrum** [comment, marked as answer]
@ShangChien : try calling `Chem.SanitizeMol()` on the output of the enumeration code:
```
m=Chem.MolFromSmiles('c1ccncc1.*:ccnc:* |m:6:0,11:5|',sanitize=False)
bundle=[Chem.SanitizeMol(x) for x in rdMolEnumerator.Enumerate(m)]
smi=Chem.MolToSmiles(bundle[0])
```
I removed the `canonical=True` argument to `MolToSmiles()` because that is the default value.
